### PR TITLE
Mobile - Text color formatting - Adds integration tests

### PR DESCRIPTION
**Pull request by geriux**
## Description
This PR is a follow-up of the main implementation https://github.com/WordPress/gutenberg/pull/36028. To add a few integration tests.

## How has this been tested?
CI check should pass

## Screenshots 
N/A

## Types of changes
Integration tests

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/format-library/src/text-color/inline.native.js [hunk 1]
```diff
@@ -148,6 +148,7 @@ export default function InlineColorUI( { name, value, onChange, onClose } ) {
 			contentStyle={ { paddingLeft: 0, paddingRight: 0 } }
 			hasNavigation
 			leftButton={ null }
+			testID="inline-text-color"
```
Comment (fluiddot) [inline packages/format-library/src/text-color/inline.native.js]: Nitpick: On other test ID definitions related to bottom sheets, we used the suffix `-modal` ([example](https://github.com/WordPress/gutenberg/blob/54049045c46380d0d05f54761fd792fe6faef4f9/packages/editor/src/components/editor-help/index.native.js#L87)) to identify that it's associated to a modal.

File: packages/format-library/src/text-color/test/index.native.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+		// Look for the pink color button
+		const pinkColorButton = await waitFor( () =>
+			getByA11yHint( COLOR_PINK )
+		);
+		expect( pinkColorButton ).toBeDefined();
+		fireEvent.press( pinkColorButton );
+
+		// Dismiss the inline color modal
+		fireEvent( inlineTextColorModal, 'backdropPress' );
```
Comment (fluiddot) [inline packages/format-library/src/text-color/test/index.native.js]: I noticed that the text color is applied when pressing a color so looks like we won't really need to dismiss the modal.

File: packages/format-library/src/text-color/test/index.native.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
```
Comment (fluiddot) [inline packages/format-library/src/text-color/test/index.native.js]: Similar to the comment on the previous test, looks like we won't really need to dismiss the modal.

**Review by fluiddot — APPROVED** [review]
Awesome job @geriux, thank you so much for adding new integration tests ⭐ .

I've just added some minor comments to the code but the PR looks great to me 🎊 .

File: packages/format-library/src/text-color/inline.native.js (hunk 1 above)
Comment (geriux) [reply packages/format-library/src/text-color/inline.native.js]: Good idea, thanks! I'll update it 👍

**Pull request merged**